src/main/java/org/repodriller/persistence/csv/CSVFile.java

```diff
@@ -16,95 +16,157 @@
 
 package org.repodriller.persistence.csv;
 
-import java.io.File;
 import java.io.FileOutputStream;
 import java.io.PrintStream;
-import java.util.Arrays;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
 
 import org.apache.commons.lang3.StringEscapeUtils;
-
+import org.apache.log4j.Logger;
+import org.repodriller.RepoDrillerException;
 import org.repodriller.persistence.PersistenceMechanism;
 
+/**
+ * A CSVFile
+ *
+ * @author Mauricio Aniche
+ */
 public class CSVFile implements PersistenceMechanism {
 
 	private PrintStream ps;
 
 	private String[] header = null;
-	
-	
+
+	private boolean writable = false;
+
+	private static final Logger log = Logger.getLogger(CSVFile.class);
+
+	/* Various constructors. */
+
+	/**
+	 * A CSVFile named fileName, overwrite existing contents, no header.
+	 *
+	 * @param fileName
+	 */
 	public CSVFile(String fileName) {
-		this(fileName, false);
+		this(fileName, false, null);
 	}
 
+	/**
+	 * A CSVFile named fileName, append?, no header.
+	 *
+	 * @param fileName
+	 * @param append If true we append, else we overwrite
+	 */
 	public CSVFile(String fileName, boolean append) {
-		try {
-			ps = new PrintStream(new FileOutputStream(fileName, append));
-		} catch (Exception e) {
-			throw new RuntimeException(e);
+		this(fileName, append, null);
+	}
+
+	/**
+	 * A CSVFile named fileName, append?, print this header?
+	 *
+	 * @param fileName
+	 * @param append If true we append, else we overwrite
+	 * @param header Print if not null
+	 */
+	public CSVFile(String fileName, boolean append, String[] header){
+		open(fileName, append);
+
+		/* Emit the header. */
+		this.header = header;
+		if (header != null)	{
+			log.error("Writing header");
+			try {
+				write((Object[]) header);
+			} catch (Exception e) {
+				throw new RuntimeException(e);
+			}
 		}
 	}
-	
+
+	/* Additional weird constructors. */
```

Comment:
these constructors are here to maintain compatibility of very old versions of repodriller. We may mark them as deprecated in the future.

Reply:
Given that you've already published old versions to Maven, is there a reason to keep deprecated methods around?